rosidl_generator_cpp/resource/msg__struct.hpp.em

```diff
@@ -81,55 +82,58 @@ struct @(spec.base_type.type)_
 {
   using Type = @(spec.base_type.type)_<ContainerAllocator>;
 
-@# constructors (with and without allocator)
-@[for (alloc_type, alloc_name) in [['', ''], ['const ContainerAllocator & ', '_alloc']]]@
-  @('explicit ' if alloc_type else '')@(spec.base_type.type)_(@(alloc_type + alloc_name))
-@# generate initializer lists
-@[  if alloc_name == '_alloc']@
-// *INDENT-OFF* (prevent uncrustify from making unnecessary indents here)
-@[  end if]@
-@{
-used_alloc_name = False
-leading_colon = ':'
-for field in spec.fields:
-    # dynamic arrays require allocator if available
-    if alloc_name and field.type.is_array and \
-            (field.type.array_size is None or field.type.is_upper_bound):
-        if leading_colon == ' ':
-            print(',')
-        print('  %s %s(%s)' % (leading_colon, field.name, alloc_name), end='')
-        used_alloc_name = True
-        leading_colon = ' '
-
-    # if default value is available set it in initializer list
-    elif field.type.is_primitive_type() and field.default_value is not None:
-        cpp_value = value_to_cpp(field.type, field.default_value)
-        if leading_colon == ' ':
-            print(',')
-        print('  %s %s(%s)' % (leading_colon, field.name, cpp_value), end='')
-        leading_colon = ' '
-if leading_colon == ' ':
-    print()
-}@
-@[  if alloc_name == '_alloc']@
-// *INDENT-ON*
-@[  end if]@
-  {
-@# generate constructor body
-@[  if alloc_name and not used_alloc_name]@
-    (void)@(alloc_name);
-@[  end if]@
 @{
-for field in spec.fields:
-    # default values of dynamic arrays with allocators if available
-    if alloc_name and field.type.is_array and \
-            (field.type.array_size is None or field.type.is_upper_bound) and \
-            field.type.is_primitive_type() and field.default_value is not None:
-        cpp_value = value_to_cpp(field.type, field.default_value)
-        print('    %s = %s;' % (field.name, cpp_value))
+def init_list(alloc_name):
+    retstring = ":"
+    used_alloc = False
+    for field in spec.fields:
+        # dynamic arrays require allocator if available
+        if field.type.is_array and \
+                (field.type.array_size is None or field.type.is_upper_bound):
```

Comment:
Tip: `field.type.is_dynamic_array()` (I know this code was just copied but I think we'll gain in readability)

Reply:
Will do.